app/auth.py

```diff
@@ -0,0 +1,68 @@
+"""This module contains authentication methods used to verify the JWT token sent by clients."""
+
+import os
+from typing import Optional
+
+import jwt
+from fastapi import HTTPException, Request
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from loguru import logger
+
+ADMIN_SECRET_KEY = os.getenv('ENP_ADMIN_SECRET_KEY')
+ALGORITHM = os.getenv('ENP_ALGORITHM', 'HS256')
+ACCESS_TOKEN_EXPIRE_SECONDS = int(os.getenv('ENP_ACCESS_TOKEN_EXPIRE_SECONDS', 60))
+
+
+class JWTBearer(HTTPBearer):
+    """JWTBearer class to verify the JWT token sent by the client."""
+
+    def __init__(self, auto_error: bool = True) -> None:
+        """Initialize the JWTBearer class.
+
+        Args:
+            auto_error (bool, optional): If True, raise an HTTPException if the token is invalid or expired. Defaults to True.
+        """
+        super(JWTBearer, self).__init__(auto_error=auto_error)
+
+    async def __call__(self, request: Request) -> Optional[HTTPAuthorizationCredentials]:
+        """Override the __call__ method to verify the JWT token. A JWT token is considered valid if it is not expired, and the signature is valid.
+
+        Args:
+            request (Request): FastAPI request object
+
+        Returns:
+            Optional[HTTPAuthorizationCredentials]: HTTPAuthorizationCredentials object if the token is valid, None otherwise.
+
+        Raises:
+            HTTPException: If the token is invalid or expired
+        """
+        credentials: HTTPAuthorizationCredentials | None = await super(JWTBearer, self).__call__(request)
+        if credentials is None:
+            logger.exception('No credentials provided.')
```

Comment:
We definitely do not want to raise an exception/error here. This is a warning state, same as a 403 would represent in something like a load balancer.
```suggestion
            logger.warning('No credentials provided.')
```